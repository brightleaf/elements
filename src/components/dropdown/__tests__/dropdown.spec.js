import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { toHaveClass, toHaveAttribute } from '@testing-library/jest-dom'
import { DropDown, DropDownMenu, DropDownDivider, DropDownItem } from '../'

expect.extend({ toHaveClass, toHaveAttribute })

describe('DropDownMenu components', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(
      <DropDownMenu label="Please Select">
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          First
        </DropDownItem>
        <DropDownDivider />
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          Second
        </DropDownItem>
      </DropDownMenu>
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('dropdown')
  })
  it('should render up', () => {
    const { container } = render(
      <DropDownMenu label="Please Select" isUp>
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          First
        </DropDownItem>
        <DropDownDivider />
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          Second
        </DropDownItem>
      </DropDownMenu>
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('dropdown')
    expect(container.firstChild).toHaveClass('is-up')
  })
  it('should when the focused `is-active` class added', async () => {
    const { container } = render(
      <DropDownMenu label="Please Select">
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          First
        </DropDownItem>
        <DropDownDivider />
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          Second
        </DropDownItem>
      </DropDownMenu>
    )

    expect(container.firstChild.firstChild).toMatchSnapshot()
    const trigger = document.querySelector('.dropdown-trigger button')
    fireEvent.focus(container.firstChild.firstChild)
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-active')
    fireEvent.click(trigger)
    expect(container.firstChild).not.toHaveClass('is-active')
    fireEvent.click(trigger)
    expect(container.firstChild).toHaveClass('is-active')
    fireEvent.click(trigger)
    expect(container.firstChild).not.toHaveClass('is-active')
  })
  it('should when the trigger is clicked `is-active` class toggled', async () => {
    const { container } = render(
      <DropDownMenu label="Please Select">
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          First
        </DropDownItem>
        <DropDownDivider />
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          Second
        </DropDownItem>
      </DropDownMenu>
    )

    const trigger = document.querySelector('.dropdown-trigger button')
    fireEvent.click(trigger)
    expect(container.firstChild).toHaveClass('is-active')
    fireEvent.click(trigger)
    expect(container.firstChild).not.toHaveClass('is-active')
    fireEvent.click(trigger)
    expect(container.firstChild).toHaveClass('is-active')
    fireEvent.click(trigger)
    expect(container.firstChild).not.toHaveClass('is-active')
  })

  it('should render with the class `is-up` when isUp is passed and have the up icon', async () => {
    const { container } = render(
      <DropDownMenu label="Please Select" isUp>
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          First
        </DropDownItem>
        <DropDownDivider />
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          Second
        </DropDownItem>
      </DropDownMenu>
    )
    expect(container.firstChild).toHaveClass('is-up')
    const icon = document.querySelector('i.fas')
    expect(icon).toHaveClass('fa-angle-up')
  })
  it('should render without the class `is-up` and have the down icon', async () => {
    const { container } = render(
      <DropDownMenu label="Please Select">
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          First
        </DropDownItem>
        <DropDownDivider />
        <DropDownItem
          onClick={e => {
            e.preventDefault()
          }}
        >
          Second
        </DropDownItem>
      </DropDownMenu>
    )
    expect(container.firstChild).not.toHaveClass('is-up')
    const icon = document.querySelector('i.fas')
    expect(icon).toHaveClass('fa-angle-down')
  })
})
