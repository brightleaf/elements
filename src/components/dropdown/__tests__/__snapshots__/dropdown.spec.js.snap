// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownMenu components should render 1`] = `
<div>
  <div
    class="dropdown"
  >
    <div
      class="dropdown-trigger"
    >
      <button
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        class="button"
      >
        <span>
          Please Select
        </span>
        <span
          class="icon is-small"
        >
          <i
            aria-hidden="true"
            class="fas fa-angle-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu animated is-hidden"
    >
      <div
        class="dropdown-content"
      >
        <a
          class="dropdown-item"
        >
          <span>
            First
          </span>
        </a>
        <hr
          class="dropdown-divider"
        />
        <a
          class="dropdown-item"
        >
          <span>
            Second
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DropDownMenu components should render up 1`] = `
<div>
  <div
    class="dropdown is-up"
  >
    <div
      class="dropdown-trigger"
    >
      <button
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        class="button"
      >
        <span>
          Please Select
        </span>
        <span
          class="icon is-small"
        >
          <i
            aria-hidden="true"
            class="fas fa-angle-up"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu animated is-hidden"
    >
      <div
        class="dropdown-content"
      >
        <a
          class="dropdown-item"
        >
          <span>
            First
          </span>
        </a>
        <hr
          class="dropdown-divider"
        />
        <a
          class="dropdown-item"
        >
          <span>
            Second
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DropDownMenu components should when the focused \`is-active\` class added 1`] = `
<div
  class="dropdown-trigger"
>
  <button
    aria-controls="dropdown-menu"
    aria-haspopup="true"
    class="button"
  >
    <span>
      Please Select
    </span>
    <span
      class="icon is-small"
    >
      <i
        aria-hidden="true"
        class="fas fa-angle-down"
      />
    </span>
  </button>
</div>
`;

exports[`DropDownMenu components should when the focused \`is-active\` class added 2`] = `
<div>
  <div
    class="dropdown is-active"
  >
    <div
      class="dropdown-trigger"
    >
      <button
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        class="button"
      >
        <span>
          Please Select
        </span>
        <span
          class="icon is-small"
        >
          <i
            aria-hidden="true"
            class="fas fa-angle-down"
          />
        </span>
      </button>
    </div>
    <div
      class="dropdown-menu animated fadeIn"
    >
      <div
        class="dropdown-content"
      >
        <a
          class="dropdown-item"
        >
          <span>
            First
          </span>
        </a>
        <hr
          class="dropdown-divider"
        />
        <a
          class="dropdown-item"
        >
          <span>
            Second
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
